tests/check/check_h26xsigned_auth.c

```diff
@@ -1712,165 +1713,82 @@ static void validation_helper_function(signed_video_t *sv, nalu_list_item_t *sei
-  SignedVideoCodec codec = settings[_i].codec;
-  nalu_list_item_t *i_nalu = nalu_list_item_create_and_set_id("I", 0, codec);
-  nalu_list_item_t *sei = NULL;
-  signed_video_t *sv_camera = NULL;
-
-  sv_camera = generate_and_set_private_key_on_camera_side(settings[_i], true, i_nalu, &sei);
-
-  // On validation side
-  signed_video_t *sv_vms = signed_video_create(codec);
-
-  // Set public key
-  sv_rc = signed_video_set_public_key(
-      sv_vms, sv_camera->signature_info->public_key, sv_camera->signature_info->public_key_size);
-  ck_assert_int_eq(sv_rc, SV_OK);
-
-  validation_helper_function(sv_vms, sei, false, sv_camera->signature_info, codec);
-
-  signed_video_free(sv_camera);
-}
-END_TEST
-
-/* Test description */
-START_TEST(no_public_key)
-{
-  // This test runs in a loop with loop index _i, corresponding to struct sv_setting _i in
-  // |settings|; See signed_video_helpers.h.
-
-  SignedVideoCodec codec = settings[_i].codec;
-  nalu_list_item_t *i_nalu = nalu_list_item_create_and_set_id("I", 0, codec);
-  nalu_list_item_t *sei = NULL;
-  signed_video_t *sv_camera = NULL;
-
-  // On camera side
-  sv_camera = generate_and_set_private_key_on_camera_side(settings[_i], false, i_nalu, &sei);
-
-  // On validation side
-  signed_video_t *sv_vms = signed_video_create(codec);
-
-  validation_helper_function(sv_vms, sei, false, sv_camera->signature_info, codec);
-
-  signed_video_free(sv_camera);
-}
-END_TEST
-
-/* Test description */
-START_TEST(public_key_on_validation_side_later)
-{
-  // This test runs in a loop with loop index _i, corresponding to struct sv_setting _i in
-  // |settings|; See signed_video_helpers.h.
-
-  SignedVideoCodec codec = settings[_i].codec;
-  nalu_list_item_t *i_nalu = nalu_list_item_create_and_set_id("I", 0, codec);
-  nalu_list_item_t *sei = NULL;
-  signed_video_t *sv_camera = NULL;
-
-  // On camera side
-  sv_camera = generate_and_set_private_key_on_camera_side(settings[_i], false, i_nalu, &sei);
-
-  // On validation side
-  signed_video_t *sv_vms = signed_video_create(codec);
-
-  validation_helper_function(sv_vms, sei, false, sv_camera->signature_info, codec);
-
-  signed_video_free(sv_camera);
-}
-END_TEST
-
-/* Test description */
-START_TEST(public_key_in_sei_and_on_validation_side_later)
-{
-  // This test runs in a loop with loop index _i, corresponding to struct sv_setting _i in
-  // |settings|; See signed_video_helpers.h.
-
-  SignedVideoCodec codec = settings[_i].codec;
-  nalu_list_item_t *i_nalu = nalu_list_item_create_and_set_id("I", 0, codec);
-  nalu_list_item_t *sei = NULL;
-  signed_video_t *sv_camera = NULL;
-
-  // On camera side
-  sv_camera = generate_and_set_private_key_on_camera_side(settings[_i], true, i_nalu, &sei);
-
-  // On validation side
-  signed_video_t *sv_vms = signed_video_create(codec);
-
-  validation_helper_function(sv_vms, sei, false, sv_camera->signature_info, codec);
-
-  signed_video_free(sv_camera);
-}
-END_TEST
+typedef struct pk_setting {
```

Comment:
I think you do not need to `typedef` it. You can even define it inside the test you use.